src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -82,7 +82,7 @@ internal sealed class CompletionSource : IAsyncExpandingCompletionSource
         _asyncListener = asyncListener;
         _editorOptionsService = editorOptionsService;
         _isDebuggerTextView = textView is IDebuggerTextView;
-        _roles = textView.Roles.ToImmutableHashSet();
+        _roles = [.. textView.Roles];
```

Comment:
>Roles [](http://example.com/codeflow?start=30&length=5)

Another. Going to stop reviewing until I can piece together how this works.

Reply:
sharplab doesn't think it's smart, but that's based on roslyn from a couple months ago, and things are definitely changing

Reply:
there's a space here for more smarts.  specifically, if there are extensions like `ToImmutableHashSet<T>(this IEnumerable<T>)` (which is what was called before, we likely should have collection-builder attributes telling the compiler it can use it.  I'll bring this up with rikki and chuck.